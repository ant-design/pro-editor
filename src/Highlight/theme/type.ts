export const THEME_DARK = 'dark';
export const THEME_LIGHT = 'light';

// 主题类型
const tuple = <T extends string[]>(...args: T) => args;
const ThemeTypes = tuple(THEME_DARK, THEME_LIGHT);
export type ThemeType = (typeof ThemeTypes)[number];
