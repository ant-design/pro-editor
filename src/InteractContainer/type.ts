import type { ReactNode } from 'react';

export type InteractionType = 'click' | 'hover';

// export enum InteractStatusEnum {
//   Hover = 'hover',
//   Unhover = 'unhover',
//   Selected = 'selected',
//   UnSelected = 'unSelected',
// }

export type InteractStatusType =
  | 'hover'
  | 'unhover'
  | 'unSelected'
  | 'selected';

// export type InteractStatusType = typeof InteractStatusEnum[keyof typeof InteractStatusEnum];
/**
 * 激活的节点
 */
export interface InteractStatusNode {
  index: number;
  rule: CanvasInteractRule;
}

/**
 * 交互规则
 */
export interface CanvasInteractRule {
  /**
   * 规则id
   */
  id: string;
  /**
   * 用于选择对象
   * 支持使用 类、id 选择器
   */
  selectors: string[];
  /**
   * 支持的交互行为
   */
  actions: InteractionType[];
}

export type ContainerNode = Document | Element;
/**
 * 交互模型
 */
export interface CanvasInteractModel extends CanvasInteractRule {
  elements: Element[];
  /**
   * 容器是否匹配
   */
  containerMatched: boolean;
}

export interface ValidElement {
  /**
   * 元素
   */
  element: HTMLElement;
  /**
   * 选择器
   */
  selector: string;
  /**
   * 序号
   */
  index: number;
}

export interface InteractStatus {
  /**
   * 交互状态的动作
   */
  action?: InteractionType;
  /**
   * 交互状态的动作
   */
  status?: InteractStatusType;
  /**
   * 当前的交互节点
   */
  currentNode?: InteractStatusNode;
  /**
   * 实际激活的节点
   */
  activeNode?: InteractStatusNode;
}

export type GetContainer = (() => HTMLElement) | HTMLElement | string;

export type OnInteractionStatusChange = (status: InteractStatus) => void;

export interface ContextCanvasProps {
  /**
   * 交互规则
   */
  rules: CanvasInteractRule[];
  /**
   * 受控模式的状态
   */
  status?: InteractStatus;
  /**
   * 状态变更回调
   */
  onStatusChange?: OnInteractionStatusChange;
  /**
   * 禁用交互
   */
  disabled?: boolean;
  /**
   * 响应画布式交互的父节点
   */
  getContainer?: GetContainer;
  /**
   * 前缀
   */
  prefixCls?: string;
  /**
   * 子组件
   */
  children?: ReactNode;
}
