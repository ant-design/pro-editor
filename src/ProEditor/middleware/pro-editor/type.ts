import { StateCreator } from 'zustand/vanilla';
import { ConfigPublicAction } from '../../store/slices/config';
import { TakeTwo, Write } from '../types/utils';

export interface ProEditorSetStateAction {
  type: unknown;

  /**
   * 如果记录，那么历史记录的名字是什么
   */
  name?: string;
  /**
   * 是否将操作记录到历史记录中
   */
  recordHistory?: boolean;
}

/**
 * 用于给注入方法添加额外的 proEditor 实例对象
 */
export interface ProEditorMiddlewareInjectMethod<Sa extends unknown[], Sr, T, A> {
  setState<A1 extends string | ProEditorSetStateAction>(...a: [...a: TakeTwo<Sa>, action?: A1]): Sr;

  proEditor: {
    undo?: () => void;
    redo?: () => void;
    // clearStorage: () => void;
    getOptions: () => Partial<ProEditorOptions<T, A>>;
  };
}
/**
 * 提供给用户的配置项
 */
export interface ProEditorOptions<S, EditorSaveState = S> {
  /** Name of the storage (must be unique) */
  name: string;
  /**
   * Filter the persisted value.
   *
   * @params state The state's value
   */
  partialize?: (state: S) => EditorSaveState;
}

// 为  mutator 注入 'pro-editor' 类型，以支持第三个配置参数

type WithProEditor<S, A> = S extends {
  getState: () => infer T;
  setState: (...a: infer Sa) => infer Sr;
}
  ? Write<S, ProEditorMiddlewareInjectMethod<Sa, Sr, T, A>>
  : never;

declare module 'zustand/vanilla' {
  interface StoreMutators<S, A> {
    ['pro-editor']: WithProEditor<S, A>;
  }
}

// 内部方法，用于保证中间件内部的类型定义安全

export type ProEditorImpl = <T>(
  storeInitializer: StateCreator<T, [['pro-editor', unknown]], []>,
  options: ProEditorOptions<T, T>,
) => StateCreator<T, [['pro-editor', unknown]], []>;

export interface InjectInternalProEditor {
  proEditor: {
    __INTERNAL_SET_CONFIG__NOT_USE_IT: ConfigPublicAction['setConfig'];
  };
}
