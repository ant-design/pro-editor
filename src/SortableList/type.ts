import type { UniqueIdentifier } from '@dnd-kit/core';
import type { ReactNode } from 'react';

export interface KeyManager {
  /**
   * 列表 ID 项
   */
  keys: UniqueIdentifier[];
  /**
   * 当前自增 ID
   */
  id: number;
}

export interface CreatorButtonProps {
  /**
   * 列表有值时是否展示添加按钮
   */
  showInList?: boolean;
  /**
   * 空数据时是否展示添加按钮
   */
  showInEmpty?: boolean;
  /**
   * 生成初始值逻辑
   */
  record?: (index: number) => Record<string, any>;
  /**
   * 新增一行按钮文案
   */
  creatorButtonText?: string;
}

export type RenderItemProps<T = any> = (item: T, index: number) => ReactNode;

export type RenderActionProps<T = any> =
  | ((item: T, index: number) => ReactNode[])
  | React.ReactNode[];

export type { UniqueIdentifier };
