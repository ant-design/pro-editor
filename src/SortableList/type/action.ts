// 新增节点
export interface AddItemAction {
  /**
   * 动作类型
   */
  type: 'addItem';
  /**
   * 新增的节点
   */
  item: any;
  /**
   * 新增节点的位置
   * @default undefined
   */
  index?: number;
}

// 移动节点
export interface MoveItemAction {
  /**
   * 动作类型
   */
  type: 'moveItem';
  /**
   * 当前节点索引
   */
  activeIndex: number;
  /**
   * 目标节点索引
   */
  overIndex: number;
}

// 移除节点
export interface RemoveItemAction {
  /**
   * 动作类型
   */
  type: 'removeItem';
  /**
   * 要移除的节点的位置
   */
  index: number;
}

// 修改节点
export interface UpdateItemAction {
  /**
   * 动作类型
   */
  type: 'updateItem';
  /**
   * 要修改的节点的位置
   */
  index: number;
  /**
   * 修改后的节点内容
   */
  item: any;
}

/**
 * 内部 Item 数据更新方法
 */
export type SortableListDispatchPayload =
  | MoveItemAction
  | AddItemAction
  | RemoveItemAction
  | UpdateItemAction;
