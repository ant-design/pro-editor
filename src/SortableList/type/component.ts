import type { DraggableSyntheticListeners, UniqueIdentifier } from '@dnd-kit/core';
import type { Transform } from '@dnd-kit/utilities';
import type { CSSProperties, FC, ReactElement, Ref } from 'react';
import { StoreUpdaterProps } from '../type/store';

export interface BaseItemProps<T = any>
  extends Pick<StoreUpdaterProps<T>, 'renderItem' | 'renderContent' | 'actions' | 'hideRemove'> {
  dragOverlay?: boolean;
  color?: string;
  disabled?: boolean;
  dragging?: boolean;
  handle?: boolean;
  height?: number;
  item: T;
  index?: number;
  fadeIn?: boolean;
  transform?: Transform | null;
  listeners?: DraggableSyntheticListeners;
  sorting?: boolean;
  className?: string;
  style?: React.CSSProperties;
  transition?: string | null;
  id: UniqueIdentifier;
  onRemove?: () => void;
}

export type RenderActionProps<T = any> =
  | ((item: T, index: number) => React.ReactNode[])
  | React.ReactNode[];

export interface SortableItemProps<T = any>
  extends Pick<
    StoreUpdaterProps<T>,
    'renderItem' | 'renderContent' | 'actions' | 'getItemStyles' | 'hideRemove' | 'handle'
  > {
  disabled?: boolean;
  id: UniqueIdentifier;
  index: number;
  item: T;
  useDragOverlay?: boolean;
  onRemove?: (index: number) => void;
}

export type RenderContent<T = any> = (item: T, index: number) => React.ReactNode;

export type RenderEmpty = () => ReturnType<FC>;

export type RenderHeader = () => ReturnType<FC>;

export interface CreatorButtonProps {
  /**
   * 添加创建按钮的位置
   */
  position?: 'bottom' | 'top';
  /**
   * 生成初始值逻辑
   */
  record: (index: number) => any;
  /**
   * 新增一行按钮文案
   */
  creatorButtonText?: string;
  /**
   * 按钮的样式设置，可以设置按钮是否显示，比如 style: {display: 'none'}
   */
  style?: CSSProperties;
}

export type RenderItem<T = any> = (
  item: T,
  options: {
    /**
     * 是否是被拖出的列表项
     */
    dragOverlay: boolean;
    /**
     * 是否在拖拽中
     */
    dragging: boolean;
    /**
     * 是否在排序中
     */
    sorting: boolean;
    /**
     * 当前列表项索引
     */
    index: number | undefined;
    /**
     * fade 动画
     */
    fadeIn: boolean;
    /**
     * 拖拽项监听器
     */
    listeners: DraggableSyntheticListeners;
    /**
     * ref
     */
    ref: Ref<HTMLElement>;
    /**
     * 当面列表项传入样式
     */
    style: CSSProperties | undefined;
    /**
     * 当面列表项 transform 动画
     */
    transform: any;
    /**
     * 当面列表项 transition 动画
     */
    transition: any;
  },
) => ReactElement;

export type { UniqueIdentifier };

interface GetItemStylesArgs {
  /**
   * 当前列表项索引
   */
  index: number;
  /**
   * 是否在拖拽中
   */
  isDragging: boolean;
  /**
   * 当前列表项 ID
   */
  id: UniqueIdentifier;
  /**
   * 是否在排序中
   */
  isSorting: boolean;
  /**
   * 拖拽覆盖的列表项索引
   */
  overIndex: number;
  /**
   * 是否是拖拽中的列表项
   */
  isDragOverlay: boolean;
}

export type GetItemStyles = (status: GetItemStylesArgs) => CSSProperties;
