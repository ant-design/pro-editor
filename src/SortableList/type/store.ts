import { ForwardedRef } from 'react';
import type {
  CreatorButtonProps,
  GetItemStyles,
  RenderActionProps,
  RenderContent,
  RenderEmpty,
  RenderItem,
  SortableListDispatchPayload,
  UniqueIdentifier,
} from '../type';

export type OnChange<T = any> = (items: T[], event: SortableListDispatchPayload) => void;

export interface SortableListState {
  /*
   * 值变化
   */
  onChange?: OnChange;
  /**
   * 删除回调
   */
  onRemove?: (index: number) => void;
  /**
   * 当前选中项 ID
   */
  activeId?: UniqueIdentifier;
  /**
   * 值
   */
  value?: any[];
  /**
   * 和 Value 值对应的 id 管理器
   */
  keyManager?: UniqueIdentifier[];
  /**
   * 渲染额外区域
   */
  actions?: RenderActionProps;
  /**
   * 渲染可排序项
   */
  renderItem?: RenderItem;
  /**
   * 渲染空状态
   */
  renderEmpty?: RenderEmpty;
  /**
   * 渲染可排序项内容
   */
  renderContent?: RenderContent;
  /**
   * 新建对象相关属性
   */
  creatorButtonProps?: CreatorButtonProps | false;
  /**
   * 渲染可排序列表项样式
   */
  getItemStyles?: GetItemStyles;
  /**
   * 隐藏
   */
  hideRemove?: boolean;
}

/**
 * 供外部使用的 ref 方法
 */
export interface SortableListRef<T = any> {
  addItem: (item?: T, index?: number) => void;
  removeItem: (index: number) => void;
  updateItem: (item: T, index: number) => void;
}

export interface StoreUpdaterProps<T = any> {
  /**
   * 值
   */
  value?: T[];
  /**
   * 初始值
   */
  initialValues?: T[];
  /**
   * 值变化
   */
  onChange?: OnChange<T>;
  /**
   * 渲染可排序项
   */
  renderItem?: RenderItem<T>;
  /**
   * 渲染内容区域
   */
  renderContent?: RenderContent<T>;
  /**
   * 渲染空状态
   */
  renderEmpty?: RenderEmpty;
  /**
   * 新建对象相关属性
   */
  creatorButtonProps?: CreatorButtonProps | false;
  /**
   * 除列表自带操作之外的其他操作自渲染
   */
  actions?: RenderActionProps<T>;
  /**
   * 对外部暴露方法
   */
  ref?: ForwardedRef<SortableListRef<T>>;
  /**
   * 如果为 true，则会在 devtools 中显示 SortableTree 内部的数据结构
   * @internal
   */
  SHOW_STORE_IN_DEVTOOLS?: boolean;
  /**
   * 是否隐藏删除按钮，默认为 false
   */
  hideRemove?: boolean;
  /**
   * 自定义排序容器样式
   */
  getItemStyles?: GetItemStyles;
}
