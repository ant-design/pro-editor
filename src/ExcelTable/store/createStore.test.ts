import { act, renderHook } from '@testing-library/react';
import Handsontable from 'handsontable';
import type { CellChange } from 'handsontable/common';

import { createStore, initialState } from './index';

vi.mock('zustand');

const useStore = createStore();

const baseData = Handsontable.helper.createEmptySpreadsheetData(3, 4);

describe('useStore', () => {
  it('默认快照', () => {
    const { result } = renderHook(() => useStore());

    expect(result.current).toMatchSnapshot('default');
  });

  it('修改默认值', () => {
    const { result } = renderHook(() => useStore());

    act(() => {
      result.current.setDefaultData(baseData);
    });

    expect(result.current.defaultData).toEqual(baseData);
  });

  it('全部重置', () => {
    const { result } = renderHook(() => useStore());

    act(() => {
      result.current.internalUpdateData(baseData);
    });

    expect(result.current.data).toEqual(baseData);

    act(() => {
      result.current.reset();
    });
    expect(result.current.data).toEqual(initialState.data);
    expect(result.current.defaultData).toEqual(initialState.defaultData);
  });

  it('重置为默认值', () => {
    const { result } = renderHook(() => useStore());

    act(() => {
      result.current.setDefaultData(baseData);
    });
    expect(result.current.data).toEqual(initialState.data);
    expect(result.current.defaultData).toEqual(baseData);

    act(() => {
      result.current.resetData();
    });
    expect(result.current.data).toEqual(baseData);
  });

  it('修改row: updateDataByRow', () => {
    const { result } = renderHook(() => useStore());

    // 初始化基础数据
    // 更新 row
    act(() => {
      result.current.internalUpdateData(baseData);
      result.current.updateByRow(4);
    });

    expect(result.current.data).toEqual(
      Handsontable.helper.createEmptySpreadsheetData(4, 4),
    );
  });
  it('修改column: updateDataByColumn', () => {
    const { result } = renderHook(() => useStore());

    // 初始化基础数据
    // 更新 column
    act(() => {
      result.current.internalUpdateData(baseData);
      result.current.updateByColumn(8);
    });

    expect(result.current.data).toEqual(
      Handsontable.helper.createEmptySpreadsheetData(3, 8),
    );
  });

  it('更新 handsontable', () => {
    const { result } = renderHook(() => useStore());

    act(() => {
      result.current.internalUpdateData(baseData);
      result.current.handleDataChanges([[0, 0, '', '123']]);
    });
    expect(result.current.data).toEqual([
      ['123', '', '', ''],
      ['', '', '', ''],
      ['', '', '', ''],
    ]);

    act(() => {
      result.current.handleDataChanges([[0, 0, '123', '123']]);
    });
    expect(result.current.data).toEqual([
      ['123', '', '', ''],
      ['', '', '', ''],
      ['', '', '', ''],
    ]);
  });

  it('受控模式：触发onChange', () => {
    const { result } = renderHook(() => useStore());

    // 设置 onChange 方法
    const dataMap = [];
    const onDataChange = (data) => {
      dataMap.push(data);
    };

    act(() => {
      result.current.setStoreState({ onDataChange });
    });
    expect(result.current.onDataChange).toEqual(onDataChange);

    // 更新 data
    act(() => {
      result.current.internalUpdateData(baseData);
      result.current.internalUpdateData(baseData);
    });

    expect(dataMap).toEqual([baseData]);

    act(() => {
      result.current.internalUpdateData(initialState.data);
    });

    expect(dataMap).toEqual([baseData, initialState.data]);
  });

  it('受控模式：外部值更新，不触发onChange', () => {
    const { result } = renderHook(() => useStore());

    expect(result.current.data).toEqual(initialState.data);

    const dataMap = [];

    // 设置 onChange 方法
    const onDataChange = (data) => {
      dataMap.push(data);
    };
    act(() => {
      result.current.setStoreState({ onDataChange });
      result.current.outsideUpdateData(baseData);
    });

    expect(dataMap).toEqual([]);
    expect(result.current.data).toEqual(baseData);
  });
});

describe('实际使用的中遇到的用例 bug', () => {
  it('复制粘贴较长的 excel到表格中，无法正常复制', () => {
    const changes: CellChange[] = [
      [0, 0, '', '序号'],
      [0, 1, '', '作品类型'],
      [0, 2, '', '样本格式'],
      [0, 3, '', '最大容量限制'],
      [0, 4, '', '文件类型'],
      [0, 5, null, '产品定价'],
      [1, 0, '', '1'],
      [1, 1, '', '文字'],
      [1, 2, '', 'PDF'],
      [1, 3, '', '10M'],
      [1, 4, '', '文字'],
      [1, 5, null, '5元/次'],
      [2, 0, '', '2'],
      [2, 1, '', '口述'],
      [2, 2, '', 'MP3、WAV'],
      [2, 3, '', '50M'],
      [2, 4, '', '音频'],
      [2, 5, null, '10元/次'],
      [3, 0, '', '3'],
      [3, 1, '', '音乐'],
      [3, 2, '', 'MP3、WAV'],
      [3, 3, '', '50M'],
      [3, 4, '', '音频'],
      [3, 5, null, '10元/次'],
      [4, 0, '', '4'],
      [4, 1, '', '戏剧'],
      [4, 2, '', 'PDF'],
      [4, 3, '', '10M'],
      [4, 4, '', '文字'],
      [4, 5, null, '5元/次'],
      [5, 0, '', '5'],
      [5, 1, '', '曲艺'],
      [5, 2, '', 'MP4、AVI、WMV'],
      [5, 3, '', '200M'],
      [5, 4, '', '视频'],
      [5, 5, null, '20元/次'],
      [6, 0, null, '6'],
      [6, 1, null, '舞蹈'],
      [6, 2, null, 'MP4、AVI、WMV'],
      [6, 3, null, '200M'],
      [6, 4, null, '视频'],
      [6, 5, null, '20元/次'],
      [7, 0, null, '7'],
      [7, 1, null, '杂技'],
      [7, 2, null, 'MP4、AVI、WMV'],
      [7, 3, null, '200M'],
      [7, 4, null, '视频'],
      [7, 5, null, '20元/次'],
      [8, 0, null, '8'],
      [8, 1, null, '美术'],
      [8, 2, null, 'JPG、PNG、JPEG'],
      [8, 3, null, '5M'],
      [8, 4, null, '图片'],
      [8, 5, null, '5元/次'],
      [9, 0, null, '9'],
      [9, 1, null, '建筑'],
      [9, 2, null, 'PDF'],
      [9, 3, null, '10M'],
      [9, 4, null, '图片'],
      [9, 5, null, '5元/次'],
      [10, 0, null, '10'],
      [10, 1, null, '摄影'],
      [10, 2, null, 'JPG、PNG、JPEG'],
      [10, 3, null, '5M'],
      [10, 4, null, '图片'],
      [10, 5, null, '5元/次'],
      [11, 0, null, '11'],
      [11, 1, null, '视听'],
      [11, 2, null, 'MP4、AVI、WMV'],
      [11, 3, null, '200M'],
      [11, 4, null, '视频'],
      [11, 5, null, '20元/次'],
      [12, 0, null, '12'],
      [12, 1, null, '图形作品（工程设计图、产品设计图、地图、示意图）'],
      [12, 2, null, 'JPG、PNG、JPEG'],
      [12, 3, null, '5M'],
      [12, 4, null, '图片'],
      [12, 5, null, '5元/次'],
    ];
    const { result } = renderHook(() => useStore());

    expect(result.current.data).toEqual(initialState.data);
    act(() => {
      result.current.handleDataChanges(changes);
    });

    expect(result.current.data).toMatchSnapshot();
  });
});
