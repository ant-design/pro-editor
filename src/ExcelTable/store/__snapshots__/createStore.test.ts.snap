// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useStore > 默认快照 > default 1`] = `
{
  "data": [
    [
      "",
      "",
      "",
      "",
      "",
    ],
    [
      "",
      "",
      "",
      "",
      "",
    ],
    [
      "",
      "",
      "",
      "",
      "",
    ],
    [
      "",
      "",
      "",
      "",
      "",
    ],
    [
      "",
      "",
      "",
      "",
      "",
    ],
    [
      "",
      "",
      "",
      "",
      "",
    ],
  ],
  "defaultData": [
    [
      "",
      "",
      "",
      "",
      "",
    ],
    [
      "",
      "",
      "",
      "",
      "",
    ],
    [
      "",
      "",
      "",
      "",
      "",
    ],
    [
      "",
      "",
      "",
      "",
      "",
    ],
    [
      "",
      "",
      "",
      "",
      "",
    ],
    [
      "",
      "",
      "",
      "",
      "",
    ],
  ],
  "handleDataChanges": [Function],
  "internalUpdateData": [Function],
  "onDataChange": null,
  "outsideUpdateData": [Function],
  "reset": [Function],
  "resetData": [Function],
  "setDefaultData": [Function],
  "setStoreState": [Function],
  "updateByColumn": [Function],
  "updateByRow": [Function],
}
`;

exports[`实际使用的中遇到的用例 bug > 复制粘贴较长的 excel到表格中，无法正常复制 1`] = `
[
  [
    "序号",
    "作品类型",
    "样本格式",
    "最大容量限制",
    "文件类型",
    "产品定价",
  ],
  [
    "1",
    "文字",
    "PDF",
    "10M",
    "文字",
    "5元/次",
  ],
  [
    "2",
    "口述",
    "MP3、WAV",
    "50M",
    "音频",
    "10元/次",
  ],
  [
    "3",
    "音乐",
    "MP3、WAV",
    "50M",
    "音频",
    "10元/次",
  ],
  [
    "4",
    "戏剧",
    "PDF",
    "10M",
    "文字",
    "5元/次",
  ],
  [
    "5",
    "曲艺",
    "MP4、AVI、WMV",
    "200M",
    "视频",
    "20元/次",
  ],
  [
    "6",
    "舞蹈",
    "MP4、AVI、WMV",
    "200M",
    "视频",
    "20元/次",
  ],
  [
    "7",
    "杂技",
    "MP4、AVI、WMV",
    "200M",
    "视频",
    "20元/次",
  ],
  [
    "8",
    "美术",
    "JPG、PNG、JPEG",
    "5M",
    "图片",
    "5元/次",
  ],
  [
    "9",
    "建筑",
    "PDF",
    "10M",
    "图片",
    "5元/次",
  ],
  [
    "10",
    "摄影",
    "JPG、PNG、JPEG",
    "5M",
    "图片",
    "5元/次",
  ],
  [
    "11",
    "视听",
    "MP4、AVI、WMV",
    "200M",
    "视频",
    "20元/次",
  ],
  [
    "12",
    "图形作品（工程设计图、产品设计图、地图、示意图）",
    "JPG、PNG、JPEG",
    "5M",
    "图片",
    "5元/次",
  ],
]
`;
